### PR TITLE
ci: add workflows & checks to PRs & Pushes

### PR DESCRIPTION
**Pull request by yusufadell**
Creating workflows and checks for the repository. The following workflows have bee added:

[flake8](https://flake8.pycqa.org/en/latest/)
[black](https://black.readthedocs.io/en/stable/index.html)
[isort](https://pycqa.github.io/isort/)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    linter:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Code Repository
+              uses: actions/checkout@v2
+
+            - name: Set up Python 3.9
+              uses: actions/setup-python@v2
+              with:
+                  python-version: 3.9
```
Comment (toolifelesstocode) [inline .github/workflows/ci.yaml:20]:
```suggestion
                  python-version: 3.8
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        branches: ["master", "main"]
+
+jobs:
+    linter:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Code Repository
+              uses: actions/checkout@v2
+
+            - name: Set up Python 3.9
```
Comment (toolifelesstocode) [inline .github/workflows/ci.yaml:17]:
```suggestion
            - name: Set up Python 3.8
```

**Review by toolifelesstocode — CHANGES_REQUESTED** [review]
Thanks so much for this! I just have some very minor change requests then it looks good to me

**Comment by yusufadell** [comment]
> Thanks so much for this! I just have some very minor change requests then it looks good to me. The minimum Python version you need is Python 3.8 for this bot so please can you just change those.

Done!

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+    pull_request:
+        branches: ["master", "main"]
```
Comment (ooliver1) [inline .github/workflows/ci.yaml:5]: Why is this locked to branches?

File: tox.ini [hunk 4]
```diff
@@ -1,2 +1,2 @@
 [flake8]
-max-line-length = 120
\ No newline at end of file
+max-line-length = 120
```
Comment (ooliver1) [inline tox.ini:2]: This file can probably be removed as it is duplicate config of `setup.cfg`

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,15 @@
+[flake8]
+max-line-length = 120
+exclude = .tox,.git,docs,venv,.venv
+
+[isort]
```
Comment (ooliver1) [inline setup.cfg:5]: Could isort use `format = "black`? Most of this config can then be removed like wrap and `use_parentheses`

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (yusufadell) [reply .github/workflows/ci.yaml:5]: Here CI/workflow makes sure that we follow a consistent coding style. and doing so for not yet merges branches is quite redundant work.

**Review by toolifelesstocode — APPROVED** [review]
Besides Oliver's requests, LGTM

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (ooliver1) [reply .github/workflows/ci.yaml:5]: The branch is always main or master, it can’t be both at the same time

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**